app/backend/src/couchers/i18n/i18n.py

```diff
@@ -0,0 +1,132 @@
+    for component_dir in i18n_dir.iterdir():
+        if not component_dir.is_dir() or component_dir.name.startswith("_"):
+            continue
+
+        component_name = component_dir.name
+        locales_dir = component_dir / "locales"
+
+        if not locales_dir.exists():
+            continue
+
+        # Load all locale JSON files for this component
+        for locale_file in locales_dir.glob("*.json"):
+            lang = locale_file.stem  # e.g., "en" from "en.json"
+
+            with open(locale_file, "r", encoding="utf-8") as f:
+                translations = json.load(f)
+
+            # Initialize nested dictionaries if needed
+            if lang not in ALL_LANGS_ALL_STRINGS:
+                ALL_LANGS_ALL_STRINGS[lang] = {}
+            if component_name not in ALL_LANGS_ALL_STRINGS[lang]:
+                ALL_LANGS_ALL_STRINGS[lang][component_name] = {}
+
+            # Store the translations
+            ALL_LANGS_ALL_STRINGS[lang][component_name] = translations
+
+    # Apply fallbacks: English is our source of truth - must exist
+    if "en" not in ALL_LANGS_ALL_STRINGS:
+        raise RuntimeError("English translations must be loaded")
+
+    en_strings = ALL_LANGS_ALL_STRINGS["en"]
+
+    # Get all languages we need to process (loaded languages + those in fallback config)
+    all_languages = set(ALL_LANGS_ALL_STRINGS.keys()) | set(LANGUAGE_FALLBACKS.keys())
+
+    for lang in all_languages:
+        if lang == "en":
+            continue  # English is already complete
+
+        # Start with a complete copy of English as the base
+        lang_strings = {}
+        for component in en_strings:
+            lang_strings[component] = en_strings[component].copy()
+
+        # Get fallback chain for this language
+        fallback_chain = LANGUAGE_FALLBACKS.get(lang, DEFAULT_FALLBACK)
+
+        # Apply fallbacks in reverse order (so more specific overrides less specific)
+        # For pt-BR with fallbacks ["pt", "en"]: Apply "en" (already done) → then "pt"
+        for fallback_lang in reversed(fallback_chain):
+            if fallback_lang in ALL_LANGS_ALL_STRINGS:
+                for component in ALL_LANGS_ALL_STRINGS[fallback_lang]:
+                    if component not in lang_strings:
+                        lang_strings[component] = {}
+                    lang_strings[component].update(ALL_LANGS_ALL_STRINGS[fallback_lang][component])
+
+        # Finally, apply the language's own translations (highest priority)
+        if lang in ALL_LANGS_ALL_STRINGS:
+            for component in ALL_LANGS_ALL_STRINGS[lang]:
+                if component not in lang_strings:
+                    lang_strings[component] = {}
+                lang_strings[component].update(ALL_LANGS_ALL_STRINGS[lang][component])
+
+        # Replace the language entry with the complete version
+        ALL_LANGS_ALL_STRINGS[lang] = lang_strings
+
+
+# Load translations and apply fallbacks when the module is imported
+_load_translations()
```

Comment:
Maybe a better approach would be to apply `functools.cache` to the function and avoid mutating global variables? And if you want to do this on startup maybe call it from `create_main_server`